### PR TITLE
Restructure readme

### PR DESCRIPTION
**Pull request by eatonphil**
Changes

* Move history sections of README into docs/HISTORY.md
* Move zig setup, testing and vopr bits into docs/HACKING.md
* Replace basic run examples with docker (so the reader doesn't need so many local dependencies)
* Show basic interaction with Node client and Node CLI

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,59 +8,53 @@ Watch an introduction to TigerBeetle on [Zig SHOWTIME](https://www.youtube.com/w
 
 [![A million financial transactions per second in Zig](https://img.youtube.com/vi/BH2jvJ74npM/0.jpg)](https://www.youtube.com/watch?v=BH2jvJ74npM)
 
-## The Problem - Realtime Processing of Balance Updates
+Read more about the [history](./docs/HISTORY.md).
```
Comment (jorangreef) [inline README.md]:
```suggestion
Read more about the [history](./docs/HISTORY.md) of TigerBeetle, the problem of balance tracking at scale and the solution of a purpose-built financial accounting database.
```

To add a little more information scent about what's hiding behind the link.

**Review by jorangreef — APPROVED** [review]
Thanks @eatonphil !

File: README.md [hunk 2]
```diff
@@ -8,59 +8,53 @@ Watch an introduction to TigerBeetle on [Zig SHOWTIME](https://www.youtube.com/w
...
-
-Furthermore, these systems may achieve performance but at the expense of reliability, especially in the face of hardware failure, corruption and misdirected writes, a compromise that is unacceptable for a system-of-record for financial accounts.
-
-## The Solution - A Purpose-Built Financial Accounting Database
+## TigerBeetle (under active development)
 
-Our survey led us to conclude that, while there are mechanisms available to shard account balances to parallelize updates and improve performance, there are still significant performance gains to be had in designing a database that is purpose built for storing balances and processing updates in a reliable manner.
+The production version of **TigerBeetle is now under active development**. Our [DESIGN doc](docs/DESIGN.md) provides an overview of TigerBeetle's data structures and our [project board](https://github.com/coilhq/tigerbeetle/projects?type=classic) provides a glimpse of where we want to go.
```
Comment (jorangreef) [inline README.md]:
```suggestion
TigerBeetle is not yet production-ready. The production version of **TigerBeetle is now under active development**. Our [DESIGN doc](docs/DESIGN.md) provides an overview of TigerBeetle's data structures and our [project board](https://github.com/coilhq/tigerbeetle/projects?type=classic) provides a glimpse of where we want to go.
```

File: README.md [hunk 3]
```diff
@@ -8,135 +8,62 @@ Watch an introduction to TigerBeetle on [Zig SHOWTIME](https://www.youtube.com/w
...
-```
-
-Run the TigerBeetle binary to see all command line arguments:
-
-```bash
-./tigerbeetle --help
-```
+* [Running a 3-node cluster locally with docker-compose](./docs/DOCKER_COMPOSE.md)
 
 ## Clients
```
Comment (sentientwaffle) [inline README.md:147]: There's also https://github.com/coilhq/tigerbeetle-go now.
Also, I noticed (below this) there's a line that says

> [client.zig](./src/vr/client.zig) is a TigerBeetle Zig client.


...which is true but maybe a little misleading. `client.zig` is `StateMachine`-agnostic — it is lower level than `tigerbeetle-node`.

In other words, `client.zig` doesn't know anything about Accounts or Transfers, only Messages. It is a library for building a client implementation, but not something to use on its own.

(Also, the path is `/vsr`, not `/vr`).

Comment (eatonphil) [reply README.md:147]: Dropped the Zig client and added the Go client. Thank you!

File: README.md [hunk 4]
```diff
@@ -2,153 +2,178 @@
...
-After ProtoBeetle, from September through October 2020, we knuckled down and rewrote TigerBeetle in C/Zig to create the alpha version of TigerBeetle, using [io_uring](https://kernel.dk/io_uring.pdf) as a foundation for fast I/O.
-
-TigerBeetle's Zig implementation of io_uring was [submitted](https://github.com/ziglang/zig/pull/6356) for addition to the Zig standard library.
-
-**[Watch a presentation of TigerBeetle given to the Interledger community on 25 November 2020.](https://www.youtube.com/watch?v=J1OaBRTV2vs)**
-
-## BetaBeetle - High Availability
-
-BetaBeetle, the beta distributed version of TigerBeetle, was developed from January 2021 through August 2021, for strict serializability, fault tolerance and automated leader election with the pioneering [Viewstamped Replication](http://pmg.csail.mit.edu/papers/vr-revisited.pdf) and consensus protocol, plus the CTRL protocol from [Protocol-Aware Recovery for Consensus-Based Storage](https://www.youtube.com/watch?v=fDY6Wi0GcPs).
+Read more about the [history](./docs/HISTORY.md) of TigerBeetle, the problem of balance tracking at scale and the solution of a purpose-built financial accounting database.
```
Comment (sentientwaffle) [inline README.md]:
```suggestion
Read more about the [history](./docs/HISTORY.md) of TigerBeetle, the problem of balance tracking at scale, and the solution of a purpose-built financial accounting database.
```

File: README.md [hunk 5]
```diff
@@ -2,153 +2,178 @@
...
-./tigerbeetle format --cluster=0 --replica=0 0_0.tigerbeetle
-./tigerbeetle format --cluster=0 --replica=1 0_1.tigerbeetle
-./tigerbeetle format --cluster=0 --replica=2 0_2.tigerbeetle
+[Join the TigerBeetle community in Discord.](https://discord.com/invite/uWCGp46uG5)
 
-./tigerbeetle start --addresses=3001,3002,3003 0_0.tigerbeetle
-./tigerbeetle start --addresses=3001,3002,3003 0_1.tigerbeetle
-./tigerbeetle start --addresses=3001,3002,3003 0_2.tigerbeetle
-```
+*If you encounter any benchmark errors, please send us the resulting `benchmark.log`.*
```
Comment (sentientwaffle) [inline README.md]: This is in the wrong section.

Comment (eatonphil) [reply README.md]: thanks! fixed

**Review by sentientwaffle — APPROVED** [review]

**Pull request merged**